### ISSUE TITLE
Selecting a slice of an alignment

### ISSUE THREAD

**Issue by GavinHuttley** [cogent3/ensembl_cli#41]
To get an alignment, we need the block_id. And to get this we need a species identifier and genome coordinate. (Let's call this `ref_coord`.) We then use `AlignDb.get_records_matching()` does this.

With the `block_id` we can then get all the coordinates for other species in
the block. (NOTE: if we are going to support strict, i.e. only blocks
where all nominated species are present, we would filter at this level).

The `gap_spans` that are returned by block will exceed the coordinates for the selected region. These will need to be trimmed to just the region so only the desired sequences need be selected. So we need to implement a container class that handles gap positions, lets call this `GapPositions`. From the `ref_coord`, we need to `GapPositions` to convert the sequence indices into alignment indices 

```python
align_start = <ref gap pos>.from_seq_to_align_index(ref_start)
align_end = align_start + ref_end - ref_start
```

For all other records, we then use  to
gives us the coordinates for that species.

```python
other_start = <other species gap pos>.from_align_to_seq_index(align_start)
other_end = <other species gap pos>.from_align_to_seq_index(align_end)
```

These values can the be used to obtain the genome sequence followed by a subset of the annotation data and combined into an `Alignment` instance.

**Comment by GavinHuttley** [comment]
The task is to implement `GapPositions.__getitem__()`

**Comment by GavinHuttley** [comment]
fixed in 279abcd
